src/main/java/com/github/manolo8/darkbot/utils/login/LoginUtils.java

```diff
@@ -167,6 +186,16 @@ private static String replaceParameters(String params) {
         return params;
     }
 
+    private static String getLoginUrl(String in) {
+        Matcher match = LOGIN_PATTERN.matcher(in);
+        if (match.find()){
+            String login_pattern = match.group(1).replace("&amp;", "&");
+            System.out.println(login_pattern);
+            return login_pattern;
+        }
```

Comment:
```suggestion
        Matcher match = LOGIN_PATTERN.matcher(in);
        if (match.find()) return match.group(1).replace("&amp;", "&");
```

never use underscores in local variable names, also no need for this print